robottelo/hosts.py

```diff
@@ -1416,11 +1415,16 @@ def register_to_cdn(self, pool_ids=None, enable_proxy=False):
             raise ContentHostError(
                 f'Error during registration, command output: {cmd_result.stdout}'
             )
-        cmd_result = self.subscription_manager_attach_pool(pool_ids)[0]
-        if cmd_result.status != 0:
-            raise ContentHostError(
-                f'Error during pool attachment, command output: {cmd_result.stdout}'
-            )
+        # Attach a pool only if the Org isn't SCA yet
+        sub_status = self.subscription_manager_status().stdout
+        if SM_OVERALL_STATUS['disabled'] not in sub_status:
+            if pool_ids is None:
```

Comment:
```
if pool_ids in [None, []]:
```
Might save us from an eventual `IndexError` in case someone passes empty list.
Also, I realized, we don't verify the type of the passed value.
adding it to the beginning of the method could allow us to bail out fast.

Reply:
I can add this check, it makes sense. This and maybe one to check the passed value is a HEX number?
I intended to make as few changes as possible to make this PR clear/straightforward and avoid curious questions. :)
Anyway, the pools are going away soon with SCA, so... we can add better validations but they will be in unreachable code section.

Reply:
You don't have to, that's why I gave you an ACK

Reply:
Hah, too late! They are in now. :smiley: